config/routes.rb

```diff
@@ -1,17 +1,36 @@
 Rails.application.routes.draw do
+  resources :qa, only: [:index] do
+    collection do
+      get 'report_bug', to: 'qa#report_bug'
+      post 'create_bug', to: 'developer#create_bug'
+      get 'show_projects', to: 'qa#show_projects'
+    end
+  end
+  resources :projects do
+    member do
+      get 'delete', action: :destroy
+    end
+  end
   devise_for :users, controllers: {
+    confirmations: 'users/confirmations',
+    passwords: 'users/passwords',
+    registrations: 'users/registrations',
     sessions: 'users/sessions',
-    registrations:'users/registrations'
+    unlocks: 'users/unlocks'
   }
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
-  #For users devise
   root to: "projects#index"
-  resources :bugs
-  resources :projects 
-      # manager
-  resources :users, only: [:show]
-      # bugs
-  resources :bugs, only: [:new, :create, :show]
-     #project-related-actions
-  resources :projects, only: [:show, :edit, :update, :create, :index]
-end
+  resources :bugs do
+    member do
+      post 'assign_to_myself'
+      post 'pick_up'
+      post 'mark_resolved'
+    end
+  end
+  resources :projects do
```

Comment:
there seems to be repetition of the resources :projects line. If you have already defined it once, you don't need to define it again with a member block.